crates/turbopack-tests/tests/snapshot/css/chained-attributes/output/crates_turbopack-tests_tests_snapshot_css_chained-attributes_input_7c9d25._.css.map

```diff
@@ -1,12 +1,12 @@
 {
   "version": 3,
   "sections": [
-    {"offset": {"line": 4, "column": 0}, "map": {"version":3,"sources":["/turbopack/[project]/crates/turbopack-tests/tests/snapshot/css/chained-attributes/input/c.css"],"sourcesContent":[".imported {\n  color: red;\n}\n"],"names":[],"mappings":"AAAA,CAAC,QAAQ,CAAC,CAAC;EACT,KAAK,EAAE,GAAG;AACZ,CAAC"}},
-    {"offset": {"line": 6, "column": 1}, "map": {"version":3,"sources":[],"names":[],"mappings":"A"}},
-    {"offset": {"line": 15, "column": 0}, "map": {"version":3,"sources":["/turbopack/[project]/crates/turbopack-tests/tests/snapshot/css/chained-attributes/input/b.css"],"sourcesContent":["@import url(\"./c.css\") layer(foo) (orientation: landscape);\n\n.imported {\n  color: orange;\n}\n"],"names":[],"mappings":"AAEA,CAAC,QAAQ,CAAC,CAAC;EACT,KAAK,EAAE,MAAM;AACf,CAAC"}},
-    {"offset": {"line": 17, "column": 1}, "map": {"version":3,"sources":[],"names":[],"mappings":"A"}},
-    {"offset": {"line": 26, "column": 0}, "map": {"version":3,"sources":["/turbopack/[project]/crates/turbopack-tests/tests/snapshot/css/chained-attributes/input/a.css"],"sourcesContent":["@import url(\"./b.css\") supports(font-format(woff2));\n\n.imported {\n  color: cyan;\n}\n"],"names":[],"mappings":"AAEA,CAAC,QAAQ,CAAC,CAAC;EACT,KAAK,EAAE,IAAI;AACb,CAAC"}},
-    {"offset": {"line": 28, "column": 1}, "map": {"version":3,"sources":[],"names":[],"mappings":"A"}},
-    {"offset": {"line": 34, "column": 0}, "map": {"version":3,"sources":["/turbopack/[project]/crates/turbopack-tests/tests/snapshot/css/chained-attributes/input/style.css"],"sourcesContent":["@import url(\"./a.css\") layer(layer) supports(not(display: inline-grid)) print;\n\n.style {\n  color: yellow;\n}\n"],"names":[],"mappings":"AAEA,CAAC,KAAK,CAAC,CAAC;EACN,KAAK,EAAE,MAAM;AACf,CAAC"}},
-    {"offset": {"line": 36, "column": 1}, "map": {"version":3,"sources":[],"names":[],"mappings":"A"}}]
+    {"offset": {"line": 4, "column": 0}, "map": {"version":3,"sources":["[project]/crates/turbopack-tests/tests/snapshot/css/chained-attributes/input/c.css"],"names":[],"mappings":"A"}},
```

Comment:
`sourcesContent` is missing in SourceMaps

Reply:
I tried debugging with `dbg`, but it seems like the source map produced by `lightningcss` is relatively minimal at the first place. 

cc @devongovett Can you take a look?

Reply:
For sourcesContent, you need to call `source_map.set_source_content` for each source index to add it. Lightning CSS doesn't store a reference or copy to the original code so that's up to you. I see [here](https://github.com/vercel/turbo/blob/b61c98e970cc7a39fbaf439997cc156afad17b08/crates/turbopack-css/src/process.rs#L134) where you add the source files but I couldn't find where the source content gets set so maybe that's missing? Or you could get the content some other way [here](https://github.com/vercel/turbo/blob/b61c98e970cc7a39fbaf439997cc156afad17b08/crates/turbopack-css/src/process.rs#L833).

As for the mappings themselves, I'm not sure without debugging it myself, but I see a couple possibilities:

1. Maybe try adding the sources before calling `to_css` instead of after?
2. When converting the source map to your internal format, maybe check that the source index is passed through (right now it is always `None`). Looks like maybe you can use `add_raw` instead. Not sure if that is related. At least verify that there are mappings there. https://github.com/vercel/turbo/blob/b61c98e970cc7a39fbaf439997cc156afad17b08/crates/turbopack-css/src/process.rs#L837-L844

The mappings are probably more sparse than you might expect though. Basically there's one per rule since that's the level of granularity that browser dev tools operate at, so more is unnecessary. But it shouldn't be empty at least.